### PR TITLE
CASSANDRA-16601 Flaky CassandraIndexTest

### PR DESCRIPTION
**Pull request by bereng**

### CODE REVIEW

File: test/unit/org/apache/cassandra/index/internal/CassandraIndexTest.java [hunk 1]
```diff
@@ -570,33 +570,39 @@ public void indexCorrectlyMarkedAsBuildAndRemoved() throws Throwable
...
-        int initialSize = rs.size();
+
+        // Wait for any background index clearing tasks to complete. Warn: When we used to run tests in parallel there
+        // could also be cross test class talk and have other indices pop up here.
+        Util.spinAssertEquals(0, () -> {
+            try
+            {
+                return execute(selectBuiltIndexesQuery).size();
+            }
+            catch(Throwable e)
```
Comment (adelapena) [inline test/unit/org/apache/cassandra/index/internal/CassandraIndexTest.java]: Super nit: missed whitespace
```suggestion
            catch (Throwable e)
```

File: test/unit/org/apache/cassandra/index/internal/CassandraIndexTest.java [hunk 2]
```diff
@@ -570,33 +570,39 @@ public void indexCorrectlyMarkedAsBuildAndRemoved() throws Throwable
...
+            {
+                fail(e.getMessage());
+            }
+            return -1;
```
Comment (adelapena) [inline test/unit/org/apache/cassandra/index/internal/CassandraIndexTest.java]: Nit: I think that `return -1;` will never be reached, it probably can be slightly simplified by just directly throwing an assertion error:
```suggestion
                throw new AssertionError(e);
            }
```

**Review by adelapena — APPROVED** [review]
Looks good to me, +1. I'm just leaving a minimal suggestion that can be addressed during commit if you agree to it.

File: test/unit/org/apache/cassandra/index/internal/CassandraIndexTest.java (hunk 2 above)
Comment (bereng) [reply test/unit/org/apache/cassandra/index/internal/CassandraIndexTest.java]: I'll be happy to take these and I'll commit.

**Comment by bereng** [comment]
CI [here](https://app.circleci.com/pipelines/github/bereng/cassandra/260/workflows/b35ae7c1-a080-495f-bf2b-c486ae1ee842)

**Pull request closed without merging**